### PR TITLE
Implement circuit breaker

### PR DESCRIPTION
**Pull request by sebtomba**
## Overview
Implement a circuit breaker with a default message size limit 200MB.

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-2792

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### CODE REVIEW

File: comm/src/main/scala/coop/rchain/comm/transport/StreamHandler.scala [hunk 1]
```diff
@@ -22,6 +22,10 @@ object StreamHandler {
 
   type CircuitBreaker = Long => Boolean
 
+  sealed trait StreamFault
+  final case class StreamError(message: String)      extends StreamFault
```
Comment (rabbitonweb) [inline comm/src/main/scala/coop/rchain/comm/transport/StreamHandler.scala]: please change those names, they are meaning less

Comment (sebtomba) [reply comm/src/main/scala/coop/rchain/comm/transport/StreamHandler.scala]: @rabbitonweb you didn't propose any better names I should use instead ;)

Comment (sebtomba) [reply comm/src/main/scala/coop/rchain/comm/transport/StreamHandler.scala]: A stream error is when the other side did something wrong. It's a user-side mistake. On the other hand, a stream exception is an unexpected error. So it's a server-side problem. Both errors together form a disjoint stream fault algebra.

Comment (itegulov) [reply comm/src/main/scala/coop/rchain/comm/transport/StreamHandler.scala]: I agree that the error names make little sense. You can probably take `CommError`'s approach here and split user-level errors (`StreamError`) into different classes (e.g. `WrongNetworkId`, `CircuitWasBroken`). Also, it seems to me that server-side errors mostly come from unexpected IO errors so you could introduce `StreamUnexpectedIOError` or something similar.

**Pull request merged**